deeplearning4j-scaleout/spark/dl4j-spark-parameterserver/src/main/java/org/deeplearning4j/spark/parameterserver/training/SharedTrainingWorker.java

```diff
@@ -0,0 +1,154 @@
+package org.deeplearning4j.spark.parameterserver.training;
+
+import lombok.Getter;
+import org.apache.spark.broadcast.Broadcast;
+import org.deeplearning4j.berkeley.Pair;
+import org.deeplearning4j.nn.conf.ComputationGraphConfiguration;
+import org.deeplearning4j.nn.conf.MultiLayerConfiguration;
+import org.deeplearning4j.nn.graph.ComputationGraph;
+import org.deeplearning4j.nn.multilayer.MultiLayerNetwork;
+import org.deeplearning4j.spark.api.TrainingHook;
+import org.deeplearning4j.spark.api.TrainingWorker;
+import org.deeplearning4j.spark.api.WorkerConfiguration;
+import org.deeplearning4j.spark.api.stats.SparkTrainingStats;
+import org.deeplearning4j.spark.api.worker.NetBroadcastTuple;
+import org.deeplearning4j.spark.impl.paramavg.BaseTrainingWorker;
+import org.deeplearning4j.spark.impl.paramavg.ParameterAveragingTrainingResult;
+import org.deeplearning4j.spark.impl.paramavg.ParameterAveragingTrainingWorker;
+import org.deeplearning4j.spark.parameterserver.conf.SharedTrainingConfiguration;
+import org.nd4j.linalg.dataset.api.DataSet;
+import org.nd4j.linalg.dataset.api.MultiDataSet;
+
+/**
+ * @author raver119@gmail.com
+ */
+public class SharedTrainingWorker extends BaseTrainingWorker<SharedTrainingResult> implements TrainingWorker<SharedTrainingResult> {
+
+    @Getter private final Broadcast<NetBroadcastTuple> broadcastModel;
+    @Getter private final Broadcast<SharedTrainingConfiguration> broadcastConfiguration;
+
+    public SharedTrainingWorker(Broadcast<NetBroadcastTuple> broadcastModel, Broadcast<SharedTrainingConfiguration> broadcastConfiguration) {
+        // our initial model is stored here.
+        this.broadcastModel = broadcastModel;
+        this.broadcastConfiguration = broadcastConfiguration;
+    }
+
+    @Override
+    public void removeHook(TrainingHook trainingHook) {
+
+    }
+
+    @Override
+    public void addHook(TrainingHook trainingHook) {
+
```

Comment:
TrainingHook doesn't *really* fit the design here, though do we want to throw an UnsupportedOperationException? I'm just worried about people trying to use it and not knowing it's a no-op here.